netcdf4/src/main/java/ucar/nc2/jni/netcdf/Nc4Iosp.java

```diff
@@ -2339,8 +2350,10 @@ private void createGroup(Group4 g4) throws IOException {
 
     // dimensions
     for (Dimension dim : g4.g.getDimensions()) {
-      int dimid = addDimension(g4.grpid, dim.getShortName(), dim.getLength());
+      int dimLength = dim.isUnlimited() ? Nc4prototypes.NC_UNLIMITED : dim.getLength();
+      int dimid = addDimension(g4.grpid, dim.getShortName(), dimLength);
```

Comment:
I'm confused - I didn't add the dimension using `addDimension`, I added it using `addUnlimitedDimension`.  Does `addUnlimitedDimension` set the length to zero by default? If I try using:

~~~java
boolean isShared = true;
boolean isUnlimited = true;
boolean isVlen = false;
ncFile.addDimension(group, "x", 10, isShared, isUnlimited, isVlen);
~~~

then I get an error, which is not what I expected. Is that the bug? Sorry, Friday brain here...